### PR TITLE
Add support for new multi-patch response and updating instance data

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

Adds support for the new multi-patch response format, and also updates the instance data if there are relevant changes. To check this we have to ignore a lot of fields like `lastChanged` (on both the instance and each dataelement in `instance.data`) which changes on every patch.

**NOTE**: The cypress tests for `multiple-datamodels-test` will fail until we get a preview version with the corresponding backend changes, as this app uses multi-patch since it has multiple datamodels.

## Related Issue(s)

- https://github.com/Altinn/app-lib-dotnet/pull/730

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [x] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Review by olemartinorg — APPROVED** [review]
Looks good! :rocket: I _might_ remove that code to compare irrelevant changes when merging with #2511, as it doesn't really matter when the instance is moved to zustand and most places won't select the entire instance data object anyway.

File: src/features/formData/FormDataWrite.tsx [hunk 1]
```diff
@@ -187,24 +189,24 @@ function useFormDataSaveMutation() {
...
-          const { newDataModels, validationIssues } = await doPatchMultipleFormData(multiPatchUrl, {
+          const { newDataModels, validationIssues, instance } = await doPatchMultipleFormData(multiPatchUrl, {
             patches,
             // Ignore validations that require layout parsing in the backend which will slow down requests significantly
             ignoredValidators: IgnoredValidators,
           });
 
           const dataModelChanges: UpdatedDataModel[] = [];
-          for (const dataElementId of Object.keys(newDataModels)) {
+          for (const { id: dataElementId, data } of newDataModels) {
```
Comment (ivarne) [inline src/features/formData/FormDataWrite.tsx:199]: The name should probably be dataElementId in the response? I think I’ll update the api

**Comment by bjosttveit** [comment]
The latest app-lib preview broke the ActionButton on `multiple-datamodels-test`, so I reverted the release to not fail elsewhere. The tests won't pass until this is fixed

**Comment by ivarne** [comment]
Action result will also receive an updated instance.

File: src/types/shared.ts [hunk 2]
```diff
@@ -64,17 +65,23 @@ export interface IDataType {
...
   instanceOwner: IInstanceOwner;
   instanceState?: IInstanceState;
   lastChanged?: string;
+  lastChangedBy?: string;
   org: string;
   selfLinks?: ISelfLinks | null;
   status?: IInstanceStatus | null;
   title?: ITitle | null;
   visibleAfter?: string;
+  process?: IProcess;
```
Comment (olemartinorg) [inline src/types/shared.ts:81]: Isn't this just deleted by `cleanUpInstanceData()`? Might be best not to trick developers into thinking the process data can be gleaned from the instance data, as the dedicated process data query fetches more stuff not returned in the instance data itself.

**Comment by olemartinorg** [comment]
I merged from main now, removed the code to compare irrelevant fields in the instance data, and (hopefully) passed on the updated instance data from action results. To prevent this change from making lots of components re-render unnecessarily, I've rewritten most of the selectors as well.

**Pull request merged**